tests/integration/targets/cloudhsm/tasks/main.yml

```
@@ -0,0 +1,109 @@
+---
+- name: 'CloudHsm Integration Tests'
+  collections:
+    - community.aws
+  module_defaults:
+    group/aws:
+      aws_access_key: '{{ aws_access_key }}'
+      aws_secret_key: '{{ aws_secret_key }}'
+      security_token: '{{ security_token | default(omit) }}'
+      region: '{{ aws_region }}'
+
+  block:
+    - name: Get AWS Subnet IDs
+      amazon.aws.ec2_vpc_subnet_info:
+      register: aws_subnets
+
+    - name: Create CloudHSM Cluster
+      community.aws.cloudhsm_cluster:
+        backup_retention_days: 90
+        subnet_ids: "{{ aws_subnets.subnets[0].subnet_id }}"
+        state: present
+        name: "{{ name }}"
+      register: new_cluster
+
+    - name: Ensure CloudHSM Cluster is Ready
+      community.aws.cloudhsm_describe_cluster:
+        name: "{{ name }}"
+      register: cluster_id
+      until: cluster_id.data[0].state == "UNINITIALIZED"
+      retries: 10
+      delay: 30
+
+    - name: Check CloudHSM Cluster Count
+      fail:
+        msg: The CloudHSM Cluster was not created
+      when: cluster_id.data|length != 1
+
+    - name: Create CloudHSM HSM Device
+      community.aws.cloudhsm_hsm:
+        state: present
+        availability_zone: "{{ aws_subnets.subnets[0].availability_zone }}"
+        cluster_id: "{{ cluster_id.data[0].cluster_id }}"
+      register: new_hsm
+
+    - name: Ensure CloudHSM HSM Device is Ready
+      community.aws.cloudhsm_describe_hsm:
```

```suggestion
      community.aws.cloudhsm_hsm_info:
```

...would be fit into the community.aws naming